### PR TITLE
Logging SDK

### PR DESCRIPTION
**Pull request by wsargent**
Setting up structure for https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3055

This work will build up a `LogEmitter` which sends out OLTP requests over GRPC/HTTP using the protobuf defined.  The Logback/Log4J appenders will then build on top of the log emitter, but other logging implementations may want to use the `LogEmitter` directly without a logging framework implementation.

Specification here https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/logs

There is an extension here https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/logging

## Implementations:

* Python: https://github.com/open-telemetry/opentelemetry-python/pull/1882
* C++: https://github.com/open-telemetry/opentelemetry-cpp/issues/356

https://aws.amazon.com/blogs/opensource/building-a-prototype-for-the-logging-api-and-sdk-for-the-opentelemetry-c-library/

## Docs

* [Google Docs](https://docs.google.com/document/d/15vR7D1x2tKd7u3zaTF0yH1WaHkUr2T4hhr7OyiZgmBg/edit#heading=h.5kjfiwmbu789)

Should talk to @tigrannajaryan for more context.

### CODE REVIEW

File: api/logging/src/main/java/io/opentelemetry/logging/LogEmitter.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.logging;
+
+public interface LogEmitter {}
```
Comment (jkwatson) [inline api/logging/src/main/java/io/opentelemetry/logging/LogEmitter.java:8]: We don't want to create a logging API at this point. Only SDK support for implementing log appenders.

**Review by kubawach — APPROVED** [review]
Looks good at this point of implementation.

**Review by kubawach — COMMENTED** [review]
At this point looks ok, awaiting some more code :)

File: docs/apidiffs/current_vs_latest/opentelemetry-context.txt [hunk 2]
```diff
@@ -1,2 +1,4 @@
 Comparing source compatibility of  against 
-No changes.
\ No newline at end of file
+***! MODIFIED INTERFACE: PUBLIC ABSTRACT io.opentelemetry.context.ContextStorage  (not serializable)
+	===  CLASS FILE FORMAT VERSION: 52.0 <- 52.0
+	---! REMOVED METHOD: PUBLIC(-) io.opentelemetry.context.Context root()
```
Comment (jkwatson) [inline docs/apidiffs/current_vs_latest/opentelemetry-context.txt:4]: ???!!!???

File: docs/apidiffs/current_vs_latest/opentelemetry-sdk-extension-jaeger-remote-sampler.txt [hunk 3]
```diff
@@ -1,2 +1,10 @@
...
\ No newline at end of file
+***! MODIFIED CLASS: PUBLIC FINAL io.opentelemetry.sdk.extension.trace.jaeger.sampler.JaegerRemoteSampler  (not serializable)
+	===  CLASS FILE FORMAT VERSION: 52.0 <- 52.0
+	---! REMOVED METHOD: PUBLIC(-) void close()
+---! REMOVED CLASS: PUBLIC(-) io.opentelemetry.sdk.extension.trace.jaeger.sampler.JaegerRemoteSamplerProvider  (not serializable)
+	---  CLASS FILE FORMAT VERSION: n.a. <- 52.0
+	---! REMOVED SUPERCLASS: java.lang.Object
+	---! REMOVED CONSTRUCTOR: PUBLIC(-) JaegerRemoteSamplerProvider()
+	---! REMOVED METHOD: PUBLIC(-) io.opentelemetry.sdk.trace.samplers.Sampler createSampler(io.opentelemetry.sdk.autoconfigure.ConfigProperties)
+	---! REMOVED METHOD: PUBLIC(-) java.lang.String getName()
```
Comment (jkwatson) [inline docs/apidiffs/current_vs_latest/opentelemetry-sdk-extension-jaeger-remote-sampler.txt:10]: looks like you might need a rebase

**Comment by ran8080** [comment]
@wsargent 
Hey, it seems like the only real point of duplication between our work is the LogAdapter class.
Also, my PR is very short and only includes the LogAdapter and Added Resource & InstrumentationLibrary to the LogRecord.
It's very small and when merged will solve some errors and failures in your one.

About the LogExporter implementations, I see you implemented **OtlpGrpcLogsExporter.java**, which is great.
We will, later on, initiate a new PR with implementation for the OtlpJsonLogsExporter, so there's not conflict here either :)

A link to it [Implement LogAdapter](https://github.com/open-telemetry/opentelemetry-java/pull/3588).
You can take a look and see :)
Let me know if it's fine with you and I'll merge.

Overall seems like a great job, and I think our code contribution will help your build and fix a few errors too.

**Comment by wsargent** [comment]
Hi @ran8080 I'm a contributor like you, not a reviewer so I'm also learning the code as I go.  Thanks for the comment!  I'll review your PR and align my code with it so it's a clean merge.

**Comment by ran8080** [comment]
@wsargent Sure thing!
I would love to be tagged at the new PR once its ready to learn from your code as well.

**Comment by jkwatson** [comment]
closing this, as we've got a functional SDK now released as alpha.

**Comment by wsargent** [comment]
Yeah, I took on more than I could chew

**Pull request closed without merging**